main/test/ca/uwaterloo/flix/library/TestSet.flix

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2016 Liam Palmer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/////////////////////////////////////////////////////////////////////////////
+// size                                                                    //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def size01: Bool = Set/size(#{}) `assertEq!` 0
+
+@test
+def size02: Bool = Set/size(#{1}) `assertEq!` 1
+
+/////////////////////////////////////////////////////////////////////////////
+// empty                                                                   //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def empty01: Bool = Set/empty() `assertEq!` #{}
+
+/////////////////////////////////////////////////////////////////////////////
+// singleton                                                               //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def singleton01: Bool = Set/singleton(1) `assertEq!` #{1}
+
+@test
+def singleton02: Bool = Set/singleton(2) `assertEq!` #{2}
+
+@test
+def singleton03: Bool = Set/singleton(true) `assertEq!` #{true}
+
+/////////////////////////////////////////////////////////////////////////////
+// insert                                                                  //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def insert01: Bool = Set/insert(1, #{}) `assertEq!` #{1}
+
+@test
+def insert02: Bool = Set/insert(1, #{1}) `assertEq!` #{1}
+/*
+@test
+def insert03: Bool = Set/insert(1, #{2}) `assertEq!` #{1, 2}
```

Comment:
As you can see from assertEq it is currently very primitive. If the two values are unequal it just fails with a generic User Exception.